pkg/gc/config.go

```diff
@@ -63,6 +70,7 @@ func NewConfigFromConfigMapFunc(ctx context.Context) func(configMap *corev1.Conf
 			cm.AsDuration("stale-revision-timeout", &c.StaleRevisionTimeout),
 			cm.AsDuration("stale-revision-lastpinned-debounce", &c.StaleRevisionLastpinnedDebounce),
 
+			cm.AsInt64("stale-revision-maximum-generations", &c.StaleRevisionMaximumGenerations),
```

Comment:
some validation below is probably in order:
0 seems invalid value (with current -1 sentinel)
all negative but -1 are invalid as well.
Also max cannot be less than min?

Reply:
0 is actually a valid value. This is a count of inactive revisions only (not including actively referenced ones). The system will never consider active revisions for deletion and setting 0 would mean "delete everything as soon as it's not serving).

Similarly a min 0 would mean "delete everything that isn't active as soon as the staleness times have elapsed"